lib/active_remote/errors.rb

```diff
@@ -42,7 +42,7 @@ def initialize(class_or_message = "")
 
   # Raised by ActiveRemove::Base.save! and ActiveRemote::Base.create! methods
   # when remote record cannot be saved because it is invalid.
-  class RemoteRecordNotSaved < ActiveRemoteError
+  class RemoteRecordNotSaved < RemoteRecordInvalid
```

Comment:
This changes the arity of the constructor since it requires an argument, which breaks the interface. Let's preserve the existing behavior (i.e., allow no message to be set) to ensure nothing breaks. We can change it in the next major version.

Reply:
Good call. I made a change and added test to cover the 3 cases: ActiveRemote object, a string message, and no message at all.